Add proposed tax estimate feature

timwis opened the pull request:
Pending OPA/Finance approval. Merge to deploy once approved. I'm out of the office on Friday 4/20.

/cc @rbrtmrtn @tswanson

timwis committed:
Add proposed tax estimate feature

timwis committed:
Use alternate tax estimator form

rmx90210 commented:
This looks nice and is working well for me. My only suggestion would be to remove the "Estimate" button, since after you click it there's no way to get it to come back. Other than that, I think this is good to go, if we ultimately decide to put this into Property.

timwis commented:
Thanks @rbrtmrtn the reason that was added was to decrease the likelihood someone might assume the number is their 2019 taxes -- it's just an estimate of the taxes based on the two rates being discussed, and doesn't account for certain tax assistance programs like OOPA (as far as I know). By making them click the "estimate" button, it (hopefully) increases the likelihood that they're aware it's an estimate. And it's a carry-over from the avi calculator.

rmx90210 committed:
Remove default homestead value

rmx90210 committed:
Update homestead exemption handling

- Handle homestead exemptions for amounts other than $30k
- Fix a bug where the proposed homestead amount of $45k was being overapplied

rmx90210 commented:
Thanks @timwis, sounds good. We were seeing a few issues with how the homestead exemption was getting applied so I pushed a few changes. I don't think I can deploy to your Surge site, so I'll create another demo and send it out.

tswanson commented:
Don't  push any updates until we get signoff on the changes that have been document

rmx90210 commented:
@tswanson Sure thing, we'll pick up next week.

rmx90210 committed:
Select proposed tax rate by default

rmx90210 committed:
Add tooltip explaning why homestead may be disabled

rmx90210 committed:
Implement revised logic for tax estimate

A few notable changes:
- When calculating proposed taxes, take the current homestead, divide by the standard homestead of $30k, and apply that ratio to the proposed homestead of $45k.
- Don't allow negative tax estimates.

timwis commented on js/views/property.js:
```diff
@@ -378,6 +378,108 @@ app.views.property = function (accountNumber) {
+      // if they already have a homestead exemption, use that value
+      if (existingHomesteadValue > 0) {
+        if (rateVal === 'proposed') {
+          displayHomesteadValue = 45000;
```
Shouldn't this be `Math.max(0, existingHomesteadValue / 30000 * 45000)`?

rmx90210 replied:
Good catch, yes -- it should reflect the proportionate, proposed homestead. I'll push a change.

rmx90210 committed:
Fix homestead label for properties with an irregular amount

rmx90210 reviewed (commented):
@timwis I updated this so it shows the proportionate amount. I don't think the `Math.max()` was necessary -- the existing homestead would have to be negative to end up with a negative proposed homestead, and I don't think there's anything like that in the data.

timwis committed:
Fix capitalisation of tax calculator, remove revenue phone number

timwis committed:
Add disclaimer to tax calculator

timwis committed:
Add link to homestead exemption page

timwis committed:
Revise disclaimer language of tax calculator
